core/modules/filter/filter.pages.inc

```diff
@@ -321,13 +338,18 @@ function filter_format_editor_image_form_submit($form, &$form_state) {
     $form_state['values']['attributes']['data-file-id'] = NULL;
   }
   // Convert any uploaded files from the FID values to the src attribute.
-  if (!empty($form_state['values']['fid'])) {
+  if (isset($form_state['values']['fid'])) {
     $fid = $form_state['values']['fid'];
-    $file = file_load($fid);
+    if ($fid == 0) {
+      $url = '/files' . $form_state['values']['image_library']['src'];
```

Comment:
If there is an external image, this results in a path like `/fileshttp://example.com/foo.png`, where `/files` is tacked to the front of a URL.

Reply:
As I ask above, is there a way we can recognise that it is an external URL and bypass this?

Reply:
Now fixed.